central/complianceoperator/v2/checkresults/service/service_impl.go

```diff
@@ -0,0 +1,120 @@
+package service
+
+import (
+	"context"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/pkg/errors"
+	complianceDS "github.com/stackrox/rox/central/complianceoperator/v2/checkresults/datastore"
+	"github.com/stackrox/rox/central/convert/storagetov2"
+	v2 "github.com/stackrox/rox/generated/api/v2"
+	"github.com/stackrox/rox/pkg/auth/permissions"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/pkg/grpc/authz"
+	"github.com/stackrox/rox/pkg/grpc/authz/perrpc"
+	"github.com/stackrox/rox/pkg/grpc/authz/user"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/search/paginated"
+	"google.golang.org/grpc"
+)
+
+const (
+	maxPaginationLimit = 1000
+)
+
+var (
+	authorizer = perrpc.FromMap(map[authz.Authorizer][]string{
+		user.With(permissions.View(resources.Compliance)): {
+			"/v2.ComplianceResultsService/GetComplianceScanResultsOverview",
+			"/v2.ComplianceResultsService/GetComplianceScanResults",
+			"/v2.ComplianceResultsService/GetComplianceProfileScanStats",
+			"/v2.ComplianceResultsService/GetComplianceClusterScanStats",
+		},
+	})
+	log = logging.LoggerForModule()
+)
+
+// New returns a service object for registering with grpc.
+func New(complianceResultsDS complianceDS.DataStore) Service {
+	return &serviceImpl{
+		complianceResultsDS: complianceResultsDS,
+	}
+}
+
+type serviceImpl struct {
+	v2.UnimplementedComplianceScanConfigurationServiceServer
+
+	complianceResultsDS complianceDS.DataStore
+}
+
+// RegisterServiceServer registers this service with the given gRPC Server.
+func (s *serviceImpl) RegisterServiceServer(grpcServer *grpc.Server) {
+	v2.RegisterComplianceResultsServiceServer(grpcServer, s)
+}
+
+// RegisterServiceHandler registers this service with the given gRPC Gateway endpoint.
+func (s *serviceImpl) RegisterServiceHandler(ctx context.Context, mux *runtime.ServeMux, conn *grpc.ClientConn) error {
+	return v2.RegisterComplianceResultsServiceHandler(ctx, mux, conn)
+}
+
+// AuthFuncOverride specifies the auth criteria for this API.
+func (s *serviceImpl) AuthFuncOverride(ctx context.Context, fullMethodName string) (context.Context, error) {
+	return ctx, authorizer.Authorized(ctx, fullMethodName)
+}
+
+// GetComplianceScanResultsOverview lists current scan configurations with most recent results overview that match the query
+func (s *serviceImpl) GetComplianceScanResultsOverview(_ context.Context, _ *v2.RawQuery) (*v2.ListComplianceScanResultsOverviewResponse, error) {
+	return nil, errox.NotImplemented
+}
+
+// GetComplianceScanResults retrieves the most recent compliance operator scan results for the specified query
```

Comment:
How do you ensure the result in the response is most recent?

Reply:
By writing ROX-20333.  I think it is difficult to correctly figure that out until we have more of the pieces pulled over.  So I punted that to later.  Which I mentioned other places.  I should throw a todo here as well.